src/server/data-access/WikibaseContentLanguagesRepo.ts

```diff
@@ -0,0 +1,66 @@
+import LanguageTranslationRepository from '@/common/data-access/LanguageTranslationRepository';
+import LanguageTranslations from '@/datamodel/LanguageTranslations';
+import TechnicalProblem from '@/common/data-access/error/TechnicalProblem';
+import LanguageRepository from '@/common/data-access/LanguageRepository';
+import LanguageCollection from '@/datamodel/LanguageCollection';
+import Language from '@/datamodel/Language';
+import MwBotWikibaseContentLanguagesRepo, { WikibaseApiContentLanguages } from './MwBotWikibaseContentLanguagesRepo';
+import { Deferred } from 'ts-deferred';
+import RtlDetectLib from 'rtl-detect';
+
+export default class WikibaseContentLanguagesRepo implements LanguageTranslationRepository, LanguageRepository {
+	private languagesRepo: MwBotWikibaseContentLanguagesRepo;
+
+	private wbContentLanguages: Deferred<WikibaseApiContentLanguages>;
+
+	public constructor( languageRepo: MwBotWikibaseContentLanguagesRepo ) {
+		this.languagesRepo = languageRepo;
+		this.wbContentLanguages = new Deferred<WikibaseApiContentLanguages>();
+	}
+
+	public loadWbContentLanguages( inLanguage: string ) {
+		this.languagesRepo.getContentLanguages( inLanguage )
+			.then( this.wbContentLanguages.resolve )
+			.catch( this.wbContentLanguages.reject );
+	}
+
+	/**
+	 * This relies on loadWbContentLanguages() to be called (through getLanguagesInLanguage())
+	 * so the actual API request is done both, in desired language and only once
+	 */
+	public getLanguages(): Promise<LanguageCollection> {
+		return this.wbContentLanguages.promise.then( ( contentLanguages: WikibaseApiContentLanguages ) => {
+			const languages = {} as LanguageCollection;
+			Object.entries( contentLanguages ).forEach( ( [ languageCode, language ] ) => {
+				languages[ languageCode ] = {
+					code: languageCode,
+					// this does not do full justice to the directionality question
+					// * languages do not have directionality - scripts do
+					// * the configured wikibase instance may have more/different languages
+					directionality: RtlDetectLib.getLangDir( languageCode ),
+				} as Language;
+			} );
+
+			return languages;
+		} );
+	}
+
+	public getLanguagesInLanguage( inLanguage: string ): Promise<LanguageTranslations> {
+		this.loadWbContentLanguages( inLanguage );
+		return this.wbContentLanguages.promise.then( ( contentLanguages: WikibaseApiContentLanguages ) => {
+			const translations = {} as any; // TODO why is multi [inLanguage][languageCode] not valid? not ltr?
```

Comment:
Not sure if I understand it correctly but the following looks cleaner IMO

Reply:
The todo is a copy from the previous incarnation of this class. If there is a cleaner way of writing this I'm happy to see it go.